lib/param_parsing/parameters.go

```diff
@@ -43,10 +54,12 @@ func GetParameters() Params {
 	getopt.StringVarLong(&params.LatestStable, "latest-stable", 's', "Latest implicit version based on a constraint. Ex: tfswitch --latest-stable 0.13.0 downloads 0.13.7 and 0.13 downloads 0.15.5 (latest)")
 	getopt.BoolVarLong(&params.ListAllFlag, "list-all", 'l', "List all versions of terraform - including beta and rc")
 	getopt.StringVarLong(&params.LogLevel, "log-level", 'g', "Set loglevel for tfswitch. One of (INFO, NOTICE, DEBUG, TRACE)")
-	getopt.StringVarLong(&params.MirrorURL, "mirror", 'm', "install from a remote API other than the default. Default: "+lib.DefaultMirror)
+
+	getopt.StringVarLong(&params.MirrorURL, "mirror", 'm', "install from a remote API other than the default. Default (based on product):\n"+strings.Join(defaultMirrors, "\n"))
 	getopt.BoolVarLong(&params.ShowLatestFlag, "show-latest", 'U', "Show latest stable version")
 	getopt.StringVarLong(&params.ShowLatestPre, "show-latest-pre", 'P', "Show latest pre-release implicit version. Ex: tfswitch --show-latest-pre 0.13 prints 0.13.0-rc1 (latest)")
 	getopt.StringVarLong(&params.ShowLatestStable, "show-latest-stable", 'S', "Show latest implicit version. Ex: tfswitch --show-latest-stable 0.13 prints 0.13.7 (latest)")
+	getopt.StringVarLong(&params.Product, "product", 'q', fmt.Sprintf("Specifies which product to use. Ex: `tfswitch --product terraform` will install Terraform. Options: (%s)", strings.Join(productIds, ", ")))
```

Comment:
Assuming Terraform is what we install by default (when this arg is not provided), I'd suggest to 1) mention this defaults to Terraform and 2) provide example with OpenTofu instead.

Also while I'm amid the review process, I'll just leave this question here (apologies if I'll find answer down the changeset from this PR): is it possible to specific default Product using TOML file? I guess this would be handy.

Reply:
> Assuming Terraform is what we install by default (when this arg is not provided), I'd suggest to 1) mention this defaults to Terraform and 2) provide example with OpenTofu instead.

Good idea - I'll do that

> Also while I'm amid the review process, I'll just leave this question here (apologies if I'll find answer down the changeset from this PR): is it possible to specific default Product using TOML file? I guess this would be handy.

Also a very good idea - I'll do this as well :)

Reply:
Apologies for the typo, I obviously meant `specify` rather than `specific`. Though this couldn't make you misinterpret for sure =) Thanks.

Reply:
> Assuming Terraform is what we install by default (when this arg is not provided), I'd suggest to 1) mention this defaults to Terraform and 2) provide example with OpenTofu instead.

This is now done

> Also while I'm amid the review process, I'll just leave this question here (apologies if I'll find answer down the changeset from this PR): is it possible to specific default Product using TOML file? I guess this would be handy.

TODO

Reply:
Configuration in TOML file has been added in 147fd939ebbf3d7890128d997e0a1e19f98d0a93